quinn-proto/src/connection/streams.rs

```diff
@@ -809,6 +818,40 @@ impl Streams {
     fn flow_blocked(&self) -> bool {
         self.data_sent >= self.max_data || self.unacked_data >= self.send_window
     }
+
+    /// Returns whether a `MAX_DATA` frame should be emitted
+    ///
+    /// If `true`, a `MAX_DATA` frame should be enqueued as soon as possible.
+    /// This will be only be the case if the window update would be significant
+    /// enough. As soon as a window update with a `MAX_DATA` frame had been
+    /// queued, the [`record_sent_max_data`] function should be called to
+    /// suppress sending further updates until the window increases significantly
+    /// again.
+    pub fn want_send_max_data(&mut self) -> bool {
+        // We never need to send invalid windows
+        if self.local_max_data > VarInt::MAX.into_inner() {
+            return false;
+        }
+
+        // Only announce a window update if it's significant enough
+        // to make it worthwhile sending a MAX_DATA frame.
+        // We use here a fraction of the configured connection receive window to make
```

Comment:
"We use a fraction of the configured connection receive window to make the decision,
to accomodate for connections using bigger windows requiring fewer updates."